--- configure	2014-04-27 12:31:56.000000000 +0200
+++ configure	2014-05-29 14:01:19.757794108 +0200
@@ -14855,7 +14855,7 @@
   CFLAGS="$_CFLAGS $LIBFTDI_CFLAGS"
   LIBS="$_LIBS $LIBFTDI_LIBS"
 
-  if test "$cross_compiling" = yes; then :
+  if true; then :
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: Skipping as we are cross-compiling, trying build only" >&5
 $as_echo "Skipping as we are cross-compiling, trying build only" >&6; }
